server/src/main/java/org/apache/iotdb/db/engine/StorageEngine.java

```diff
@@ -394,6 +413,15 @@ public void asyncCloseProcessor(PartialPath storageGroupPath, boolean isSeq)
         processor.writeUnlock();
       }
     } else {
+      // bug fix, for storage group do does not have data
+      // https://issues.apache.org/jira/browse/IOTDB-875
+      List<PartialPath> allStorageGroupPaths = MManager.getInstance()
```

Comment:
can you use `getStorageGroupPath` instead of getting all storage groups?

Reply:
sure, thanks for your advice.